geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -0,0 +1,327 @@
+    def regexExpr   = string <~ "::r" ^^ { case LitString(s) => RegexExpr(s) }
+    def column      = "$" ~> "[1-9][0-9]*".r ^^ { i => Col(i.toInt) }
+    def cast2int    = expr <~ "::int" ^^ { e => Cast2Int(e) }
+    def cast2double = expr <~ "::double" ^^ { e => Cast2Double(e) }
+    def fieldLookup = "$" ~> ident ^^ { i => FieldLookup(i) }
+    def fnName      = ident ^^ { n => LitString(n) }
+    def fn          = (fnName <~ OPEN_PAREN) ~ (repsep(transformExpr, ",") <~ CLOSE_PAREN) ^^ {
+      case LitString(name) ~ e => FunctionExpr(functionMap(name).build(name), e)
+    }
+    def strEq       = ("strEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => StrEQ(l, r)
+    }
+    def intEq       = ("intEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntEQ(l, r)
+    }
+    def intLTEq     = ("intLTEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntLTEQ(l, r)
+    }
+    def intLT       = ("intLT" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntLT(l, r)
+    }
+    def intGTEq     = ("intGTEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntGTEQ(l, r)
+    }
+    def intGT       = ("intGT" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntGT(l, r)
+    }
+    def dEq       = ("dEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DEQ(l, r)
+    }
+    def dLTEq     = ("dLTEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DLTEQ(l, r)
+    }
+    def dLT       = ("dLT" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DLT(l, r)
+    }
+    def dGTEq     = ("dGTEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DGTEQ(l, r)
+    }
+    def dGT       = ("dGT" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DGT(l, r)
+    }
+    def binaryPred  = strEq | intEq | intLTEq | intLT | intGTEq | intGT | dEq | dLTEq | dLT | dGTEq | dGT
+    def andPred     = ("and" ~ OPEN_PAREN) ~> (pred ~ "," ~ pred) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => And(l, r)
+    }
+    def orPred      = ("or" ~ OPEN_PAREN) ~> (pred ~ "," ~ pred) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => Or(l, r)
+    }
+    def notPred     = ("not" ~ OPEN_PAREN) ~> pred <~ CLOSE_PAREN ^^ {
+      case pred => Not(pred)
+    }
+    def logicPred = andPred | orPred | notPred
+    def pred: Parser[Predicate] = binaryPred | logicPred
+    def expr = fn | wholeRecord | regexExpr | fieldLookup | column | lit
+    def transformExpr: Parser[Expr] = cast2double | cast2int | expr
```

Comment:
The DSL is slick.  What about adding a "cast2string", "isNull", and conditional operator in a future version?